src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use PHPUnit\Framework\Assert;
+use Symfony\Bundle\FrameworkBundle\Client;
+
+/**
+ * Ideas borrowed from Laravel Dusk's assertions.
+ *
+ * @see https://laravel.com/docs/5.7/dusk#available-assertions
+ */
+trait WebTestAssertions
+{
+    /** @var Client|null */
+    protected static $client;
+
+    public static function assertResponseIsSuccessful(): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertTrue(
+            $response->isSuccessful(),
+            "Response was expected to be successful, but actual HTTP code is {$response->getStatusCode()}."
+        );
+    }
+
+    public static function assertHttpCodeEquals(int $expectedCode): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertSame(
+            $expectedCode,
+            $code = $response->getStatusCode(),
+            "Response code [$expectedCode] does not match actual HTTP code [$code]."
+        );
+    }
+
+    public static function assertResponseRedirects(int $expectedCode = null, string $expectedLocation = null): void
+    {
+        self::checkResponseAvailable();
+
+        $response = static::$client->getResponse();
+
+        Assert::assertTrue(
+            $response->isRedirect(),
+            "Actual response code [{$response->getStatusCode()}] does not match any valid HTTP redirection code."
+        );
+
+        if ($expectedCode) {
+            static::assertHttpCodeEquals($expectedCode);
+        }
+
+        if ($expectedLocation) {
```

Comment:
```suggestion
        if ($expectedLocation !== null) {
```

Otherwise it will silently skip the check if someone accidentally puts `''` as an `$expectedLocation`.

Reply:
A location that equals `""` is not something valid, so this should never happen

Reply:
@Pierstoval It should never happen, but it can. But in case I accidentally put there `''` as expectedLocation (e.g. when i will have a list of URLs in the testcase and I'm iterating through them) I would prefer to know it immediately instead of letting it fail silently.

Reply:
Well, I can change this so any non-null may fail in this case, seems fair